lib/routemap.c

```diff
@@ -636,13 +630,25 @@ static struct route_map *route_map_new(const char *name)
 /* Add new name to route_map. */
 static struct route_map *route_map_add(const char *name)
 {
-	struct route_map *map;
+	struct route_map *map, *exist;
 	struct route_map_list *list;
 
 	map = route_map_new(name);
 	list = &route_map_master;
 
-	/* Add map to the hash */
+	/*
+	 * Add map to the hash
+	 *
+	 * If the map already exists in the hash, then we know that
+	 * FRR is now in a sequence of delete/create.
+	 * All FRR needs to do here is set the to_be_processed
+	 * bit (to inherit from the old one
+	 */
+	exist = hash_release(route_map_master_hash, map);
+	if (exist) {
+		map->to_be_processed = exist->to_be_processed;
```

Comment:
If we're changing (re-adding) a previously existing route-map, shouldn't we set this to `true` unconditionally? The new route-map is likely different from the old one…

Reply:
each daemon has the ability to turn this on/off via the callbacks associated with it.  If they choose to immediately process and not delay then unconditionally setting this would mean nothing then.  Nor is there a methodology to get it turned off( because it is assumed to be turned off again when the timer pops and the callback is called to iterate over all the route-maps.